mount-helper-container/Makefile

```diff
@@ -16,18 +32,98 @@ BIN_DIR := bin
+	cp ${BIN_DIR}/eitserver $(BUILD_DIR)/usr/local/bin/
+	cp install/postinst.sh $(BUILD_DIR)/DEBIAN/postinst
+	cp install/prerm.sh $(BUILD_DIR)/DEBIAN/prerm
+
+	echo "Package: $(BUILD_DIR)" > $(DEBIAN_CONTROL)
+	echo "Version: $(APP_VERSION)" >> $(DEBIAN_CONTROL)
+	echo "Maintainer: $(MAINTAINER)" >> $(DEBIAN_CONTROL)
+	echo "Architecture: $(DEB_ARCH)" >> $(DEBIAN_CONTROL)
+	echo "Description: $(DESCRIPTION)" >> $(DEBIAN_CONTROL)
+	echo "Depends: mount-ibmshare-1.0.0" >> $(DEBIAN_CONTROL)
+
+	dpkg-deb --build $(BUILD_DIR)
+	rm -rf $(BUILD_DIR)
+
+rpm-build:
+rpm-build: build-linux
+	rm -rf $(BUILD_DIR)
+	mkdir -p $(BUILD_DIR)/
+	mkdir -p $(BUILD_DIR)/rpm/SOURCES/usr/local/bin $(BUILD_DIR)/rpm/SOURCES/etc/systemd/system $(BUILD_DIR)/rpm/SOURCES/etc/ibmcloud
+	cp install/mount-helper-container.service $(BUILD_DIR)/rpm/SOURCES/etc/systemd/system
+	cp install/share.conf $(BUILD_DIR)/rpm/SOURCES/etc/ibmcloud
+	cp ${BIN_DIR}/eitserver $(BUILD_DIR)/rpm/SOURCES/usr/local/bin
+
+	echo "Name: $(NAME)" > $(REDHAT_SPEC)
+	echo "Version: $(APP_VERSION)" >> $(REDHAT_SPEC)
+	echo "Release: $(RPM_RELEASE_NUM)" >> $(REDHAT_SPEC)
+	echo "Summary: $(DESCRIPTION)" >> $(REDHAT_SPEC)
+	echo "License: $(LICENSE)" >>  $(REDHAT_SPEC)
+	echo "BuildArch: $(RPM_ARCH)" >>  $(REDHAT_SPEC)
+	echo "Requires: mount-ibmshare" >> $(REDHAT_SPEC)
+	echo "%define _rpmfilename $(NAME)-$(APP_VERSION).rpm" >> $(REDHAT_SPEC)
+	echo "%build" >> $(REDHAT_SPEC)
+
+	echo "%install" >> $(REDHAT_SPEC)
+	echo "rm -rf %{buildroot}" >> $(REDHAT_SPEC)
+	echo "mkdir -p %{buildroot}/etc/systemd/system/" >> $(REDHAT_SPEC)
+	echo "mkdir -p %{buildroot}/usr/local/bin/" >> $(REDHAT_SPEC)
+	echo "mkdir -p %{buildroot}/etc/ibmcloud/" >> $(REDHAT_SPEC)
+	echo "cp -r %{_sourcedir}/* %{buildroot}/" >> $(REDHAT_SPEC)
+
+	echo "%description" >> $(REDHAT_SPEC)
+	echo "This package provides the IBM Mount Share helper container service."
+
+	echo "%files" >> $(REDHAT_SPEC)
+	echo "/etc/systemd/system/mount-helper-container.service" >> $(REDHAT_SPEC)
+	echo "/usr/local/bin/eitserver" >> $(REDHAT_SPEC)
+	echo "/etc/ibmcloud/share.conf" >> $(REDHAT_SPEC)
+
+	echo "%post" >> $(REDHAT_SPEC)
+	echo "systemctl enable mount-helper-container.service" >> $(REDHAT_SPEC)
+	echo "systemctl start mount-helper-container.service" >> $(REDHAT_SPEC)
+
+	echo "%preun" >> $(REDHAT_SPEC)
+	echo "systemctl stop mount-helper-container.service || true" >> $(REDHAT_SPEC)
+	echo "systemctl disable mount-helper-container.service || true" >> $(REDHAT_SPEC)
+
+	rpmbuild -ba --build-in-place --define "_topdir $(PWD)/$(BUILD_DIR)/rpm" $(REDHAT_SPEC)
+	cp  $(BUILD_DIR)/rpm/RPMS/* ./
+	rm -rf $(BUILD_DIR)
+
+packages:
+packages: deb-build rpm-build clean
+
 build:
-	CGO_ENABLED=0 GOOS=$(go env GOOS) GOARCH=$(go env GOARCH) go build -mod mod -o ${BIN_DIR}/mhelper-container-server ./server/server.go
+	CGO_ENABLED=0 GOOS=$(go env GOOS) GOARCH=$(go env GOARCH) go build -mod mod -o ${BIN_DIR}/eitserver ./server/server.go
```

Comment:
Any specific reason why this has been renamed from mhelper-container-server to eitserver? mhelper-container-server sounds specific to MH code while eitserver sounds more generic.

Reply:
No particular reason. I can change that

Reply:
Done